junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertIterableEquals.java

```diff
@@ -67,6 +67,11 @@ private static void assertIterableEquals(Iterable<?> expected, Iterable<?> actua
 			if (expectedElement == actualElement) {
 				continue;
 			}
+			// Prevent stack overflow error.
+			// See https://github.com/junit-team/junit5/issues/2157 for details.
+			if (expectedElement != null && expectedElement.equals(actualElement)) {
```

Comment:
```suggestion
			if (Objects.equals(expectedElement, actualElement)) {
```

You can then also merge the two `if`s into one because `Objects.equals()` already checks for `==`.

Reply:
Neat improvement.